src/wamr/WAMRWasmModule.cpp

```diff
@@ -69,35 +75,38 @@ namespace wasm {
 
         // Load the wasm file
         storage::FileLoader &functionLoader = storage::getFileLoader();
-        std::vector<uint8_t> aotFileBytes = functionLoader.loadFunctionWamrAotFile(msg);
+#if(WAMR_EXECUTION_MODE_INTERP)
+        wasmBytes = functionLoader.loadFunctionWasm(msg);
+#else
+        std::vector<uint8_t> wasmBytes = functionLoader.loadFunctionWamrAotFile(msg);
+#endif
 
-        // Load wasm
-        errorBuffer.reserve(ERROR_BUFFER_SIZE);
+        // Load wasm module
         wasmModule = wasm_runtime_load(
-                aotFileBytes.data(),
-                aotFileBytes.size(),
-                errorBuffer.data(),
+                wasmBytes.data(),
+                wasmBytes.size(),
+                errorBuffer,
                 ERROR_BUFFER_SIZE
         );
 
         if(wasmModule == nullptr) {
-            std::string errorMsg = std::string(errorBuffer.data());
-            logger->error("Failed to instantiate WAMR module: \n{}", errorMsg);
-            throw std::runtime_error("Failed to instantiate WAMR module");
+            std::string errorMsg = std::string(errorBuffer);
+            logger->error("Failed to load WAMR module: \n{}", errorMsg);
+            throw std::runtime_error("Failed to load WAMR module");
         }
         
         // Instantiate module
-        moduleInstance = wasm_runtime_instantiate(
+        if(!(moduleInstance = wasm_runtime_instantiate(
```

Comment:
In this project we don't nest function calls in if statement conditions, rather split them onto separate lines:

```
moduleInstance = wasm_runtime_instantiate(
   ...
);

if(!moduleInstance) {
...
}
```

Reply:
👍